features/support/env.rb

```diff
@@ -3,40 +3,58 @@
 require 'open4'
 require 'rspec'
 
+
 SOURCE_DIRECTORY = "#{File.dirname(__FILE__)}/../../src"
 SHELL_OVERRIDE_DIRECTORY = "#{File.dirname(__FILE__)}/shell_overrides"
 
+MEMOIZED_REPOSITORY_BASE = Dir.mktmpdir 'memoized'
 REPOSITORY_BASE = Dir.mktmpdir
 TOOLS_INSTALLED_FILENAME = "#{REPOSITORY_BASE}/tools_installed.txt"
 
-Before do
-  Dir.chdir REPOSITORY_BASE
-  FileUtils.rm_rf Dir.glob("#{REPOSITORY_BASE}/*")
 
+# load memoized environment by copying contents
+# of MEMOIZED_REPOSITORY_BASE to REPOSITORY_BASE
+def setup_environment
+  FileUtils.rm_rf REPOSITORY_BASE
+  FileUtils.cp_r "#{MEMOIZED_REPOSITORY_BASE}/.", REPOSITORY_BASE
+end
+
+
+# rubocop:disable Metrics/MethodLength
+def initialize_environment
   # Create origin repository
   create_repository :origin
 
   # Create the local repository
   clone_repository :origin, :developer
```

Comment:
This line can be removed which lets you leave rubocop on. When in_repository with an identifier, you can pass in a parent to clone from if the repository does not exist (and the default parent is `:origin`)